src/zope/securitypolicy/securitymap.py

```diff
@@ -30,7 +30,8 @@ def _clear(self):
     def __nonzero__(self):
         return bool(self._byrow)
 
-    __bool__ = __nonzero__
+    def __bool__(self):
+        return self.__nonzero__()
```

Comment:
I'm really surprised by this change and would like to understand it better.

(A regression test would help a lot!)

Reply:
I made a simple test, and yes, the previous code works under Python 2, and fails under Python 3, the new code works both for Python 2 and Python 3.